src/main/java/seedu/logic/errorchecker/MainChecker.java

```diff
@@ -73,11 +68,16 @@ public void checkGender(String stringToken) throws InvalidGenderException {
     }
 
     public void illegalCharacterChecker(String stringToken, String fieldInput) throws IllegalCharacterException {
-        String nameString = stringToken.toLowerCase();
+
+        String cleanedInput = UI.cleanseInput(stringToken);
+        if (!stringToken.equals(cleanedInput)){
+            throw new IllegalCharacterException(fieldInput);
+        }
+        /*String nameString = stringToken.toLowerCase();
         Pattern pattern = Pattern.compile(ILLEGAL_CHARACTERS);
```

Comment:
Can consider removing commented out code for a cleaner codebase. However, i understand this allows easy reversal if the current method does not work well.